src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs

```diff
@@ -52,7 +51,13 @@ protected virtual bool CanUpdateProperty(ModelMetadata propertyMetadata)
         internal async Task<bool> CanCreateModel(MutableObjectBinderContext context)
         {
             var bindingContext = context.ModelBindingContext;
+
+            // Reviewers: The isTopLevelObject value is wrong whenever a binder (e.g. CollectionModelBinder) uses
+            // GetMetadataForType(). Suggest we need ModelMetadata.GetMetadataForElementType() and a special case here
+            // for ComplexModelDto. Impact will likely be mainly extra "empty" (no bound property) objects created.
+            // Worth filing an issue?
```

Comment:
started writing up an issue and convinced myself our current binders don't use `GetMetadataForType()` enough to make this issue visible. might be worth adding `IsTopLevelObject` for protection going forward but:
- existing `isTopLevelObject` value is generally correct unless the model type is an element of a collection
- that incorrect value matters only when either the prefix is empty (fine) or the `BinderModelName` is not `null`
- the `IModelNameProvider` attributes can't be applied to a class